### PR TITLE
test web controllers

### PR DESCRIPTION
**Pull request by maksimk99**
1. add badges to readme.md
2. add tests for controllers
3. replace big constructors with setters which return object
4. create a list of dates using thymeleaf instead of java class
see #2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# carhouse-client
+# carhouse-client [![Build Status](https://travis-ci.com/java-fat-unicorn-team/carhouse-client.svg?branch=master)](https://travis-ci.com/java-fat-unicorn-team/carhouse-client) [![codecov](https://codecov.io/gh/java-fat-unicorn-team/carhouse-client/branch/master/graph/badge.svg)](https://codecov.io/gh/java-fat-unicorn-team/carhouse-client)
```
Comment (aleksieiev) [inline README.md:1]: looks like independent PR

File: README.md [hunk 2]
```diff
@@ -10,12 +10,12 @@ Project is created with:
 * Git 2+
 * Gradle 5.5+
 
-##Build
+## Build
```
Comment (aleksieiev) [inline README.md:13]: It also can be sent as independent PR

**Comment by maksimk99** [comment]
Ready for review 
@aleksieiev

File: src/main/java/com/carhouse/provider/impl/CarMakeProviderImpl.java [hunk 3]
```diff
@@ -33,7 +33,11 @@
      * @param carMakeId the car make id
      * @return the car make
      */
-    public CarMake getCarMake(final Integer carMakeId) {
-        return new CarMake(carMakeId, "BMW");
+    public CarMake getCarMake(final String carMakeId) {
+        if (carMakeId != null && Integer.parseInt(carMakeId) >= 0) {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/provider/impl/CarMakeProviderImpl.java:37]: exception possible if `carMakeId` isn't Int

File: src/main/java/com/carhouse/provider/impl/CarModelProviderImpl.java [hunk 4]
```diff
@@ -20,8 +19,12 @@
      * @param carModelId the car model id
      * @return the car model
      */
-    public CarModel getCarModel(final Integer carModelId) {
-        return new CarModel(carModelId, new CarMake(), "M2");
+    public CarModel getCarModel(final String carModelId) {
+        if (carModelId != null && Integer.parseInt(carModelId) >= 0) {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/provider/impl/CarModelProviderImpl.java:23]: exception possible if `carModelId` isn't Int

File: src/main/java/com/carhouse/provider/impl/CarModelProviderImpl.java [hunk 5]
```diff
@@ -30,13 +33,13 @@ public CarModel getCarModel(final Integer carModelId) {
      * @param carMakeId the car make id
      * @return the list car model
      */
-    public List<CarModel> getCarModels(final Integer carMakeId) {
-        if (carMakeId >= 0) {
+    public List<CarModel> getCarModels(final String carMakeId) {
+        if (carMakeId != null && Integer.parseInt(carMakeId) >= 0) {
```
Comment (aleksieiev) [inline src/main/java/com/carhouse/provider/impl/CarModelProviderImpl.java:37]: exception possible if `carMakeId` isn't Int

**Review by aleksieiev — APPROVED** [review]
minor changes required. Might be improved during the next refactoring
